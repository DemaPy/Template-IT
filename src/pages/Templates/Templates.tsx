import PageContainer from "../../components/PageContainer";
import { PlusCircle } from "lucide-react";
import CreateTemplate from "./components/CreateTemplate";
import GridView from "../../components/GridView";
import TemplateCard from "./components/TemplateCard";
import Heading from "../../components/Heading";
import { useTemplateCreateModal } from "../../store/templateCreateModal";
import { useFetchTemplates } from "./pages/hooks/useTemplate";
import toast from "react-hot-toast";
import ComponentsSkeleton from "../Components/components/Skeleton";
import Error from "../Error/Error";

const Templates = () => {
  const setIsOpen = useTemplateCreateModal((state) => state.setOpen);

  const { data, isError, error, isPending } = useFetchTemplates()

  if (isPending) return <ComponentsSkeleton />

  if (isError) {
    toast.error(error.message);
    return <Error message={error.message} path="/" />
  }

  if (!data) {
    toast.error("Unexpected error happend.");
    return
  }

  return (
    <PageContainer>
      <Heading
        title={"Templates"}
        action={{
          icon: <PlusCircle className="w-4 h-4 mr-2" />,
          onClick: setIsOpen,
          title: "create",
        }}
      />
      <GridView items={data.data} component={TemplateCard} />
      <CreateTemplate />
    </PageContainer>
  );
};

export default Templates;
