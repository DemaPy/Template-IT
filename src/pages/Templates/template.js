[
   {
      "title":"Monday",
      "sections":[
         {
            "content":"    <table\n      cellspacing=\"0\"\n      cellpadding=\"0\"\n      style=\"max-width: 650px; width: 100%; background-color: #ffccb7\"\n    >\n      <tbody>\n        <tr>\n          <td>\n            <a href=\"\">\n              <img\n                alt=\"\"\n                src=\"\"\n                style=\"vertical-align: middle; max-width: 100%\"\n                loading=\"lazy\"\n              />\n            </a>\n          </td>\n        </tr>\n        <tr>\n          <td>\n            <a href=\"\">\n              <img\n                alt=\"\"\n                src=\"\"\n                style=\"vertical-align: middle; max-width: 100%\"\n                loading=\"lazy\"\n              />\n            </a>\n          </td>\n        </tr>\n        <tr>\n          <td class=\"newsletterBottom35px\"></td>\n        </tr>\n      </tbody>\n    </table>",
            "placeholders":[
               {
                  "name":"Top image href",
                  "position":195,
                  "fallback":"https://www.beliani.ch/",
                  "id":"37e1ab4e-d35f-4266-952e-fa7648e7b342"
               },
               {
                  "name":"Top image src",
                  "position":261,
                  "fallback":"https://placehold.co/650x200",
                  "id":"c45861e7-52f3-44c3-ad40-66a0b23ab9d5"
               },
               {
                  "name":"Top gif href",
                  "position":471,
                  "fallback":"https://www.beliani.ch/",
                  "id":"2ff5c8d1-677d-45c9-ba91-fcedb356a884"
               },
               {
                  "name":"Top gif src",
                  "position":537,
                  "fallback":"https://placehold.co/650x400",
                  "id":"f2e9f03b-2a6a-4c20-a629-be11d0df1623"
               }
            ],
            "title":"Top images",
            "id":"b0633072-f939-4fbe-a4a0-ea602729cf82"
         },
         {
            "content":"    <table\n      cellspacing=\"0\"\n      cellpadding=\"0\"\n      style=\"max-width: 650px; width: 100%; background-color: #ffccb7\"\n    >\n      <tbody>\n        <tr>\n          <td align=\"center\">\n            <span class=\"newsletterTitle\"></span>\n          </td>\n        </tr>\n        <tr>\n          <td align=\"center\" class=\"newsletterBottom20px\"></td>\n        </tr>\n        <tr>\n          <td align=\"center\">\n            <span class=\"newsletterParagraph\"></span>\n          </td>\n        </tr>\n        <tr>\n          <td align=\"center\" class=\"newsletterBottom35px\"></td>\n        </tr>\n        <tr>\n          <td align=\"center\">\n            <span class=\"newsletterTitle\"></span>\n          </td>\n        </tr>\n        <tr>\n          <td align=\"center\" class=\"newsletterBottom20px\"></td>\n        </tr>\n        <tr>\n          <td align=\"center\">\n            <span class=\"newsletterParagraph\"></span>\n          </td>\n        </tr>\n        <tr>\n          <td align=\"center\" class=\"newsletterBottom35px\"></td>\n        </tr>\n        <tr>\n          <td align=\"center\">\n            <span class=\"newsletterTitle\"></span>\n          </td>\n        </tr>\n        <tr>\n          <td align=\"center\" class=\"newsletterBottom20px\"></td>\n        </tr>\n        <tr>\n          <td align=\"center\">\n            <span class=\"newsletterParagraph\"></span>\n          </td>\n        </tr>\n        <tr>\n          <td align=\"center\" class=\"newsletterBottom35px\"></td>\n        </tr>\n      </tbody>\n    </table>",
            "placeholders":[
               {
                  "name":"Title 1",
                  "position":231,
                  "fallback":"Title Fallback 1",
                  "id":"4ea37ff7-e3fc-46e0-8ab7-3815ebdfb76a"
               },
               {
                  "name":"Title 2",
                  "position":663,
                  "fallback":"Title Fallback 2",
                  "id":"42ae756c-6473-445e-bad5-7e91a4298f18"
               },
               {
                  "name":"Title 3",
                  "position":1095,
                  "fallback":"Title Fallback 3",
                  "id":"5788b45a-3d7b-4d1a-b6ba-7e4adbd78fc4"
               },
               {
                  "name":"Paragraph 1",
                  "position":449,
                  "fallback":"Paragraph Fallback 4",
                  "id":"e14e60a1-da3d-413f-a540-3909aab3e450"
               },
               {
                  "name":"Paragraph 2",
                  "position":881,
                  "fallback":"Paragraph Fallback 5",
                  "id":"acc1a8b7-4639-44b5-92fd-1734e9a05a0c"
               },
               {
                  "name":"Paragraph 3",
                  "position":1313,
                  "fallback":"Paragraph Fallback 6",
                  "id":"a302bf38-ca03-403d-876c-29c66342cf88"
               }
            ],
            "title":"Offer part",
            "id":"22281359-e66b-4fbf-b1ce-678baa9765f0"
         },
         {
            "content":"<table cellspacing=\"0\" cellpadding=\"0\" border=\"0\" align=\"center\" width=\"100%\">\n<tbody>\n<tr>\n<td>\n<table cellspacing=\"0\" cellpadding=\"0\" border=\"0\" align=\"center\" width=\"100%\">\n<tbody>\n<tr>\n<td align=\"left\" class=\"newsletterBottom35px\">\n<span class=\"newsletterTitle\">\n</span>\n</td>\n</tr>\n</tbody>\n</table>\n</td>\n</tr>\n<tr>\n<td>\n<table cellspacing=\"0\" cellpadding=\"0\" border=\"0\" align=\"center\" width=\"100%\">\n<tbody>\n<tr>\n<td align=\"left\" class=\"newsletterContainer\">\n<span class=\"newsletterParagraph\"></span>\n</td>\n</tr>\n</tbody>\n</table>\n</td>\n</tr>\n</tbody>\n</table>",
            "placeholders":[
               {
                  "name":"Paragraph",
                  "position":499,
                  "id":"09dfc18c-fa4d-406f-aff2-0973f9ae6627",
                  "fallback":"test"
               }
            ],
            "title":"Intro",
            "id":"0e7fc5b4-cf0c-4470-8e16-251d3bfefe99"
         },
         {
            "content":"    <table\n      cellspacing=\"0\"\n      cellpadding=\"0\"\n      style=\"max-width: 650px; width: 100%; background-color: \"\n    >\n      <tbody>\n        <tr>\n          <td align=\"center\">\n            <span class=\"newsletterTitle\"></span>\n          </td>\n        </tr>\n        <tr>\n          <td align=\"center\" class=\"newsletterBottom20px\"></td>\n        </tr>\n        <tr>\n          <td align=\"center\">\n            <span class=\"newsletterParagraph\"></span>\n          </td>\n        </tr>\n        <tr>\n          <td align=\"center\" class=\"newsletterBottom35px\"></td>\n        </tr>\n        <tr>\n          <td align=\"center\">\n            <span class=\"newsletterTitle\"></span>\n          </td>\n        </tr>\n        <tr>\n          <td align=\"center\" class=\"newsletterBottom20px\"></td>\n        </tr>\n        <tr>\n          <td align=\"center\">\n            <span class=\"newsletterParagraph\"></span>\n          </td>\n        </tr>\n        <tr>\n          <td align=\"center\" class=\"newsletterBottom35px\"></td>\n        </tr>\n        <tr>\n          <td align=\"center\">\n            <span class=\"newsletterTitle\"></span>\n          </td>\n        </tr>\n        <tr>\n          <td align=\"center\" class=\"newsletterBottom20px\"></td>\n        </tr>\n        <tr>\n          <td align=\"center\">\n            <span class=\"newsletterParagraph\"></span>\n          </td>\n        </tr>\n        <tr>\n          <td align=\"center\" class=\"newsletterBottom35px\"></td>\n        </tr>\n      </tbody>\n    </table>",
            "placeholders":[
               {
                  "name":"Color",
                  "position":117,
                  "id":"66f9a8f3-25db-49ef-8310-1056dd9a380a",
                  "fallback":"#ffffff"
               }
            ],
            "title":"TEST",
            "id":"0e780e86-4b23-4f19-838b-0cc201b0f28a"
         }
      ],
      "wrapper":{
         "html":"",
         "placeholders":[
            
         ]
      },
      "id":"1d2228ae-dca3-41dc-a0ef-b5f737919c6e"
   }
]