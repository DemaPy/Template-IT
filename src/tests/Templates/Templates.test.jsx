import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import { expect, it } from "vitest";
import Templates from "../../pages/Templates/Templates";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { RouterProvider, createBrowserRouter } from "react-router-dom";

afterEach(cleanup);
const queryClient = new QueryClient({
  defaultOptions: {
    mutations: {
      retry: false,
    },
  },
});

describe("Test Templates component", () => {
  const router = createBrowserRouter([
    {
      path: "/",
      element: (
        <QueryClientProvider client={queryClient}>
          <Templates />
        </QueryClientProvider>
      ),
    },
  ]);
  it("Templates component has been rendered", () => {
    render(<RouterProvider router={router} />);
    expect(screen.getByText("Templates")).toBeInTheDocument();
  });

  it("Templates component match snapshot", () => {
    const result = render(<RouterProvider router={router} />);
    expect(result).toMatchSnapshot()
  })
});
