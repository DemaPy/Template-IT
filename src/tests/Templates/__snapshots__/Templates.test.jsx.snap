// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test Templates component > Templates component match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="grow flex items-stretch flex-col space-y-6 h-full"
      >
        <div
          class="flex flex-row justify-between items-center gap-2"
        >
          <h1
            class="lg:text-4xl md:text-2xl text-xl font-semibold leading-tight text-slate-700"
          >
            Templates
          </h1>
          <button
            aria-controls="radix-:r3:"
            aria-expanded="false"
            aria-haspopup="dialog"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-white transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-zinc-950 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 dark:ring-offset-zinc-950 dark:focus-visible:ring-zinc-300 bg-zinc-900 text-zinc-50 hover:bg-zinc-900/90 dark:bg-zinc-50 dark:text-zinc-900 dark:hover:bg-zinc-50/90 h-10 px-4 py-2"
            data-state="closed"
            type="button"
          >
            <svg
              class="lucide lucide-circle-plus w-4 h-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <path
                d="M8 12h8"
              />
              <path
                d="M12 8v8"
              />
            </svg>
          </button>
        </div>
        <section
          class="grow flex items-stretch flex-col space-y-6 h-full"
        >
          <div
            class="flex flex-col gap-2"
          >
            <div
              class="animate-pulse dark:bg-zinc-800 h-[400px] p-4 flex flex-col gap-3 border rounded-md bg-blue-100 border-blue-300"
            />
            <div
              class="animate-pulse dark:bg-zinc-800 h-[400px] p-4 flex flex-col gap-3 border rounded-md bg-blue-100 border-blue-300"
            />
            <div
              class="animate-pulse dark:bg-zinc-800 h-[400px] p-4 flex flex-col gap-3 border rounded-md bg-blue-100 border-blue-300"
            />
            <div
              class="animate-pulse dark:bg-zinc-800 h-[400px] p-4 flex flex-col gap-3 border rounded-md bg-blue-100 border-blue-300"
            />
            <div
              class="animate-pulse dark:bg-zinc-800 h-[400px] p-4 flex flex-col gap-3 border rounded-md bg-blue-100 border-blue-300"
            />
            <div
              class="animate-pulse dark:bg-zinc-800 h-[400px] p-4 flex flex-col gap-3 border rounded-md bg-blue-100 border-blue-300"
            />
            <div
              class="animate-pulse dark:bg-zinc-800 h-[400px] p-4 flex flex-col gap-3 border rounded-md bg-blue-100 border-blue-300"
            />
            <div
              class="animate-pulse dark:bg-zinc-800 h-[400px] p-4 flex flex-col gap-3 border rounded-md bg-blue-100 border-blue-300"
            />
            <div
              class="animate-pulse dark:bg-zinc-800 h-[400px] p-4 flex flex-col gap-3 border rounded-md bg-blue-100 border-blue-300"
            />
            <div
              class="animate-pulse dark:bg-zinc-800 h-[400px] p-4 flex flex-col gap-3 border rounded-md bg-blue-100 border-blue-300"
            />
          </div>
        </section>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="grow flex items-stretch flex-col space-y-6 h-full"
    >
      <div
        class="flex flex-row justify-between items-center gap-2"
      >
        <h1
          class="lg:text-4xl md:text-2xl text-xl font-semibold leading-tight text-slate-700"
        >
          Templates
        </h1>
        <button
          aria-controls="radix-:r3:"
          aria-expanded="false"
          aria-haspopup="dialog"
          class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-white transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-zinc-950 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 dark:ring-offset-zinc-950 dark:focus-visible:ring-zinc-300 bg-zinc-900 text-zinc-50 hover:bg-zinc-900/90 dark:bg-zinc-50 dark:text-zinc-900 dark:hover:bg-zinc-50/90 h-10 px-4 py-2"
          data-state="closed"
          type="button"
        >
          <svg
            class="lucide lucide-circle-plus w-4 h-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <path
              d="M8 12h8"
            />
            <path
              d="M12 8v8"
            />
          </svg>
        </button>
      </div>
      <section
        class="grow flex items-stretch flex-col space-y-6 h-full"
      >
        <div
          class="flex flex-col gap-2"
        >
          <div
            class="animate-pulse dark:bg-zinc-800 h-[400px] p-4 flex flex-col gap-3 border rounded-md bg-blue-100 border-blue-300"
          />
          <div
            class="animate-pulse dark:bg-zinc-800 h-[400px] p-4 flex flex-col gap-3 border rounded-md bg-blue-100 border-blue-300"
          />
          <div
            class="animate-pulse dark:bg-zinc-800 h-[400px] p-4 flex flex-col gap-3 border rounded-md bg-blue-100 border-blue-300"
          />
          <div
            class="animate-pulse dark:bg-zinc-800 h-[400px] p-4 flex flex-col gap-3 border rounded-md bg-blue-100 border-blue-300"
          />
          <div
            class="animate-pulse dark:bg-zinc-800 h-[400px] p-4 flex flex-col gap-3 border rounded-md bg-blue-100 border-blue-300"
          />
          <div
            class="animate-pulse dark:bg-zinc-800 h-[400px] p-4 flex flex-col gap-3 border rounded-md bg-blue-100 border-blue-300"
          />
          <div
            class="animate-pulse dark:bg-zinc-800 h-[400px] p-4 flex flex-col gap-3 border rounded-md bg-blue-100 border-blue-300"
          />
          <div
            class="animate-pulse dark:bg-zinc-800 h-[400px] p-4 flex flex-col gap-3 border rounded-md bg-blue-100 border-blue-300"
          />
          <div
            class="animate-pulse dark:bg-zinc-800 h-[400px] p-4 flex flex-col gap-3 border rounded-md bg-blue-100 border-blue-300"
          />
          <div
            class="animate-pulse dark:bg-zinc-800 h-[400px] p-4 flex flex-col gap-3 border rounded-md bg-blue-100 border-blue-300"
          />
        </div>
      </section>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
